### ISSUE TITLE
compile error: using ‘typename’ outside of template

### ISSUE THREAD

**Issue by billzt** [bcgsc/arcs#25]
```
g++ -DHAVE_CONFIG_H -I. -I..  -I/home/zhut/src/arcs-1.0.1/Arcs -I/home/zhut/src/arcs-1.0.1/Common -I/home/zhut/src/arcs-1.0.1/DataLayer -I/home/zhut/src/arcs-1.0.1 -I/home/zhut/src/arcs-1.0.1   -isystem/home/zhut/src/arcs-1.0.1/1_58_0 -Wall -Wextra -Werror -std=c++0x -fopenmp -g -O2 -MT arcs-Arcs.o -MD -MP -MF .deps/arcs-Arcs.Tpo -c -o arcs-Arcs.o `test -f 'Arcs.cpp' || echo '/home/zhut/src/arcs-1.0.1/Arcs/'`Arcs.cpp
In file included from /usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../include/c++/4.4.7/backward/hash_set:60,
                 from /usr/include/boost/graph/adjacency_list.hpp:25,
                 from /usr/include/boost/graph/undirected_graph.hpp:11,
                 from Arcs.h:20,
                 from Arcs.cpp:2:
/usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../include/c++/4.4.7/backward/backward_warning.h:28:2: error: #warning This file includes at least one deprecated or antiquated header which may be removed without further notice at a future date. Please use a non-deprecated interface with equivalent functionality instead. For a listing of replacement headers and interfaces, consult the file backward_warning.h. To disable this warning use -Wno-deprecated.
In file included from Arcs.cpp:2:
Arcs.h:86: error: using ‘typename’ outside of template
Arcs.h:100: error: using ‘typename’ outside of template
In file included from Arcs.cpp:3:
../Arcs/DistanceEst.h:49: error: using ‘typename’ outside of template
../Arcs/DistanceEst.h:53: error: using ‘typename’ outside of template
../Arcs/DistanceEst.h:75: error: using ‘typename’ outside of template
../Arcs/DistanceEst.h: In function ‘void buildPairToBarcodeStats(const ARCS::IndexMap&, const std::unordered_map<std::basic_string<char, std::char_traits<char>, std::allocator<char> >, int, std::hash<std::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<const std::basic_string<char, std::char_traits<char>, std::allocator<char> >, int> > >&, const ARCS::ContigToLength&, const ARCS::ArcsParams&, PairToBarcodeStats&)’:
../Arcs/DistanceEst.h:216: error: using ‘typename’ outside of template
../Arcs/DistanceEst.h: In function ‘void addEdgeDistances(const PairToBarcodeStats&, const JaccardToDist&, const ARCS::ArcsParams&, ARCS::Graph&)’:
../Arcs/DistanceEst.h:401: error: expected initializer before ‘:’ token
../Arcs/DistanceEst.h:430: error: expected primary-expression before ‘}’ token
../Arcs/DistanceEst.h:430: error: expected ‘;’ before ‘}’ token
../Arcs/DistanceEst.h:430: error: expected primary-expression before ‘}’ token
../Arcs/DistanceEst.h:430: error: expected ‘)’ before ‘}’ token
../Arcs/DistanceEst.h:430: error: expected primary-expression before ‘}’ token
../Arcs/DistanceEst.h:430: error: expected ‘;’ before ‘}’ token
cc1plus: warnings being treated as errors
../Arcs/DistanceEst.h: At global scope:
../Arcs/DistanceEst.h:393: error: unused parameter ‘pairToStats’
../Arcs/DistanceEst.h:393: error: unused parameter ‘params’
../Arcs/DistanceEst.h:393: error: unused parameter ‘g’
../Arcs/DistanceEst.h: In function ‘void writeDistTSV(const std::string&, const PairToBarcodeStats&, const ARCS::Graph&)’:
../Arcs/DistanceEst.h:457: error: expected initializer before ‘:’ token
Arcs.cpp:1054: error: expected primary-expression at end of input
Arcs.cpp:1054: error: expected ‘;’ at end of input
Arcs.cpp:1054: error: expected primary-expression at end of input
Arcs.cpp:1054: error: expected ‘)’ at end of input
Arcs.cpp:1054: error: expected statement at end of input
Arcs.cpp:1054: error: expected ‘}’ at end of input
../Arcs/DistanceEst.h: At global scope:
../Arcs/DistanceEst.h:433: error: unused parameter ‘pairToStats’
../Arcs/DistanceEst.h:433: error: unused parameter ‘g’
make[2]: *** [arcs-Arcs.o] Error 1
make[2]: Leaving directory `/home/zhut/src/arcs-1.0.1/Arcs'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/home/zhut/src/arcs-1.0.1'
make: *** [all] Error 2
```

**Comment by billzt** [comment]
Well, my system is gcc 4.4.7 and boost 1.41 (CentOS 6). 
This problem has been resolved after I use gcc 4.9.2 and boost 1.60 (I use software collection)
Please remind it in the README

**Comment by lcoombe** [comment]
Thanks for the update, @billzt
